client/src/components/WatchlistButtonsDetailView.js

```diff
@@ -70,9 +71,87 @@ function WatchlistButtonsDetailView({ showDetails }) {
       return actor.name;
     }),
   };
-  const { data: user } = useQuery(['user', authenticatedUser.userId], getUser, {
+
+  const { data: user } = useQuery(['user', userId], getUser, {
     staleTime: 3600000,
   });
+  const [addToWatch] = useMutation(addToWatchList, {
+    onSuccess: () => {
+      queryCache.refetchQueries('user', {
+        force: true,
+      });
+      queryCache.refetchQueries('recsGenres', {
+        force: true,
+      });
+      queryCache.refetchQueries('recsNetworks', {
+        force: true,
+      });
+      queryCache.refetchQueries('toWatchList', {
+        force: true,
+      });
+      queryCache.refetchQueries('watchedList', {
+        force: true,
+      });
+    },
```

Comment:
You have the same `onSuccess` callback multiple times in this file.

```suggestion
    onSuccess: refetchQueries,
```